### DISCUSSION TITLE
BIG performance improvement in map rendering

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#434] [Q&A] [answered]
Cruiser will have a really **_BIG_** performance improvement in map rendering.

---

Test: import the large [Marguerite route](https://en.wikipedia.org/wiki/Marguerite_route), hundreds of tracks with total length 4209 km.
Make it visible and scroll the map when inside the screen.

GPX is available here: [Marguritruten.zip](https://github.com/devemux86/cruiser/files/14157759/Marguritruten.zip)

---

You will see the **_BIG_** difference in the next Cruiser: zero lag and ultra-fast performance. :rocket: 

Allowing us to load and display many large routes and tracks.

<img src="https://github.com/devemux86/cruiser/assets/3484020/6300cf4d-cab7-4b84-86fb-32d4fd51f239" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c4085e0c-07ec-4c1a-93f5-0272b938a91f" width="50%"/>

**Comment by kds112** [comment]
It sounds gorgeous... 🚀

**Comment by HannesHoBu** [comment]
Looking really forward, the next impressive update

**Comment by Ciutnik** [comment]
I have about 100.000 km saved in OziExplorer .plt files from my motorcycle trips. I can easily convert them to .gpx, but I don't think I would dare to upload the whole thing to Cruiser, but I'll be curious to see where the limit is 😀

**Reply by devemux86** [reply]
@Ciutnik After the next update it's worth doing such heavy tests. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
I did it! I imported 230 tracks with a total length of several dozen thousand kilometers! The Cruiser displays them ALL at the same time without any problem 😳 In my case, it's good that the source GPX files were deleted. This is what I wanted. But did this happen because I manually put them into the Cruiser's my-tracks folder? If they were anywhere else, would they also be deleted during import?

I gradually reduced the scale at which the tracks became visible, expecting that at some point there would be too much data to display on the map... Unnecessarily. I reached zoom level 2 and everything was still working smoothly, although there wasn't much of the map visible under the tracks (especially near my house, where the tracks start and end) 😀

@devemux86 really great job, I'm very impressed! Thank you very much! 👍

**Reply by devemux86** [reply]
@Ciutnik thanks for the tests!

The performance improvement is indeed very significant. :rocket: 

> I manually put them into the Cruiser's my-tracks folder?

The correct way to add favorites (of any kind) is to use their **import** functions.

Don't just copy files into folders, as Cruiser needs to process them on import to optimize their performance.

**Reply by Ciutnik** [reply]
> The correct way to add favorites (of any kind) is to use their import functions.

That's right. Cruiser didn't automatically notice the GPX files I put manually into the my-tracks folder, so I had to use the import function anyway. This is completely obvious (because there is probably no point in implementing automatic import in such a case) 🙂

**Reply by devemux86** [reply]
> there is probably no point in implementing automatic import in such a case

We could, as also in the case of adding manually offline maps / theme files.
There are performance reasons Cruiser should do the job, but we could see it in the future. :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
BIG performance improvement in map rendering is available in [Cruiser 3.0.39](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by umpfmpf** [comment]
Is there a limit in the route length? 
I just tried to load the Route de Grande Alps, but I get a message: Route not found. 
Only when I remove about 25-30% of the route does he calculate it.

**Reply by devemux86** [reply]
@umpfmpf this is about the **rendering** of the route / track lines on the map.

Not the route **calculation** by the routing service.
Route calculation depends on the selected route options and routing service.
What router and options have you used?

**Reply by umpfmpf** [reply]
I used the brouter offline. 
In the options I started with the motorcycle settings and then voted one option after the other. 
It is also not possible with the car and hiking profile.
With brouter online it works fine.

**Reply by devemux86** [reply]
@umpfmpf 

Please create a new topic so we can discuss this with details.
Also post the route (is it a GPX file?) and your routing options.

Better start with the default options: all avoid options are off.
Calculate the first route and then start testing more options.

**Comment by ramondevic** [comment]
I used the Cruiser option to save my routes but in one of the updates I messed everything up, I could see the name and duration in kms and time of the route but when I loaded it I got an error... luckily I had the files originals in ".kurviger" format

Right now I always have the waypoints, tracks and routes saved separately in folders available to be imported from Cruiser, after the trip I delete them from "favorites" and I can always import the original files from outside again when I need them .

**Reply by devemux86** [reply]
@ramondevic 

If you still have any of the failed files, you can send them so I can check what happens.

Do you import them with the app interface or copy files manually in the file system?
(as @Ciutnik mentioned [above](https://github.com/devemux86/cruiser/discussions/434#discussioncomment-8407443))

**Reply by ramondevic** [reply]
They were files created on kurviger.de and imported a few weeks ago to use on upcoming trips I'm preparing, I just had them on my phone to test. I have deleted them directly from Cruiser and re-imported.

I have the files on the PC, I copy them to the phone and import them from Cruiser.

Thanks !

**Comment by 1maxi2** [comment]
Verstehe ich das richtig. Die Leistungsverbesserung betrifft nur den schnelleren Kartenaufbau.
Eine schnellere Routenberechnung in brouter offline ist davon nicht betroffen.

**Reply by devemux86** [reply]
@1maxi2 

It is about the fast display of routes and tracks on the map.
Fast BRouter (offline) routing will be available with:

Es geht um die schnelle Anzeige von Routen und Tracks auf der Karte.
Schnelles BRouter (offline) Routing wird mit verfügbar sein:

- https://github.com/devemux86/cruiser/discussions/142
